txpipe/data_types/base.py

```diff
@@ -444,4 +445,28 @@ def write_provenance(self):
         pass
 
     def read_provenance(self):
-        raise ValueError("Reading existing PNG files is not supported")
\ No newline at end of file
+        raise ValueError("Reading existing PNG files is not supported")
+
+class PickleFile(DataFile):
+    suffix = "pkl"
+
+    @classmethod
+    def open(self, path, mode, **kwargs):
+        return open(path, mode + "b")
+
+    def write_provenance(self):
+        self.write(self.provenance)
+
+    def read_provenance(self):
+        return self.read()
+
+    def write(self, obj):
+        if self.mode != "w":
+            raise ValueError("Tried to write to read-only pickle file")
+        pickle.dump(obj, self.file)
+
+    def read(self):
+        if self.mode != "r":
+            raise ValueError("Tried to read from a write-only pickle file")
```

Comment:
I would add self.mode to the print statement, in order to help users in debugging (this is true in several places, and maybe a ModeError exception is in order)

Reply:
I've changed this to `io.UnsupportedError`, which is what you get if you try to do the equivalent thing with an ordinary python file object, and added a print of self.mode.

I've also added a check when opening the file in the first place that the argument is either "r" or "w".